modules/storages/app/models/storages/common/configuration_checks.rb

```diff
@@ -42,9 +42,12 @@ def configured?
     end
 
     def configuration_checks
-      { host_name_configured: (host.present? && name.present?),
-        openproject_oauth_application_configured: oauth_application.present?,
-        storage_oauth_client_configured: oauth_client.present? }
+      { storage_oauth_client_configured: oauth_client.present? }.tap do |configuration_checks|
+        if provider_type_nextcloud?
```

Comment:
Imho, it would be more simple to not have this module/concern and put the logic inside `storage.rb` base class and define/redefine `configuration_checks` in the descendants. Wdyt?

Reply:
Not exactly, before it was a set of methods that were extracted into a module, but the behavior is the same for the whole hierarchy. So, it is fine. But with this PR we are introducing the check on the type of the class inside the module which is included inside a base class. So, it smells like `configuration_checks` method is now a responsibility of descendants.

Reply:
I don't see it as a violation of responsibility.

the concern is scoping everything it needs to give you an answer about the validity of your configuration. independent of the storage provider type. putting those check into the storage class descendants would again bloat them, while having them within the concern makes it easier to grasp all differences in configuration checking within all provider types.

Reply:
But it is already dependent, because the behavior differs for each subclass inside `configurations_check`. This piece belongs there in the first place, because of the type context. And the module presence in this case just decrease the code readability, imho.

Reply:
As discussed with the devs, we will destruct the concern and integrate the check to the specific descendants.

The storage.rb spec will be dismantled into a shared_example, to be executed in `NextcloudStorage` as well as in `OneDriveStorage`. the test cases from the concern will be written explicitly for both provider types.

There will be a next PR including those changes, link will be added by edith.